assets/tests/components/ladderPage.test.tsx

```diff
@@ -525,6 +532,52 @@ describe("LadderPage", () => {
     )
     expect(result.queryByText("Vehicles")).toBeNull()
   })
+
+  test("opens the detour modal", async () => {
+    jest
+      .mocked(getTestGroups)
+      .mockReturnValue([
+        TestGroups.DetourRouteSelection,
+        TestGroups.DetoursPilot,
+        TestGroups.RouteLadderHeaderUpdate,
+      ])
+
+    const mockState = {
+      ...initialState,
+      routeTabs: [
+        routeTabFactory.build({
+          selectedRouteIds: ["1"],
+          ordering: 0,
+          isCurrentTab: true,
+        }),
+      ],
+    }
+    const { container } = render(
+      <StateDispatchProvider state={mockState} dispatch={mockDispatch}>
+        <BrowserRouter>
+          <RoutesProvider routes={routes}>
+            <LadderPage />
+          </RoutesProvider>
+        </BrowserRouter>
+      </StateDispatchProvider>
+    )
+
+    expect(container.querySelector(".c-new-route-ladder__header")).toBeVisible()
+
+    expect(
+      container.querySelector(".c-route-ladder__dropdown-button")
+    ).toBeVisible()
+
+    await userEvent.click(
+      container.querySelector(".c-route-ladder__dropdown-button") as Element
+    )
+
+    expect(screen.getByRole("button", { name: "Add detour" })).toBeVisible()
+
+    await userEvent.click(screen.getByRole("button", { name: "Add detour" }))
+
+    expect(screen.getByText("Create Detour")).toBeVisible()
```

Comment:
question(non-blocking): I'd prefer the `getByRole("heading", { name: "Create Detour" })` call here, is there a reason you wouldn't or would prefer `getByText` here?

Reply:
No - I was actually hoping to figure out the Right™ way to do this. I copied this from [the analogous mapPage.test](https://github.com/mbta/skate/blob/25c2f662b818f072b82afd7f1911795c675cc4df/assets/tests/components/mapPage.test.tsx#L1852).

Reply:
This is fixed now, but I squash-rebased and lost the specific `fixup!` commit that did this.

Reply:
I see it! [`48c6fba1c3f543a154189e273dc683f6c0507986..e47aa0764f9a0a259897026543c9194eec20744a`#diff-41b776228e](https://github.com/mbta/skate/compare/48c6fba1c3f543a154189e273dc683f6c0507986..e47aa0764f9a0a259897026543c9194eec20744a#diff-41b776228e88467a5a40b857932c4e403705be68b76270e8b50f440ec6abcb3cR577)